### PR TITLE
Single source of truth for locale information

### PR DESCRIPTION
**Pull request by biwano**
## Description

Single source of truth for locale information

If we want to have the Opportunity to have differences between app and site we can make the getLocales function parametrized.

## Related Ticket

Resolves #nothing
It is related to @Atmosfearful comment here: https://github.com/KlimaDAO/klimadao/pull/728#pullrequestreview-1125566392

## Changes

None

## Checklist

- [X] Building the site with `npm run build-site` works without errors
- [X] Building the app with `npm run build-app` works without errors
- [X] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: lib/utils/getLocales/index.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      n: number | string,
+      ord?: boolean
+    ) => "zero" | "one" | "two" | "few" | "many" | "other";
+    time: string;
+    label: string;
+    production_ready: boolean;
+  };
+}
+
+const locales: ILocales = {
```
Comment (Atmosfearful) [inline lib/utils/getLocales/index.ts:18]: how about `productionLocales` and `stagingLocales` defined as separate objects? just makes it more clear at a glance

File: lib/utils/getLocales/index.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { en, fr, de, ru, zh, ko, es, hi } from "make-plural/plurals";
```
Comment (Atmosfearful) [inline lib/utils/getLocales/index.ts:1]: We need to add this (or lingui) as a dependency for @klimadao/lib

File: lib/constants/index.ts [hunk 3]
```diff
@@ -254,3 +254,5 @@ export const verra = {
   projectSearch: `${VERRA_REGISTRY_API}/resource/resource/search?maxResults=2000&$count=true&$skip=0&$top=50`,
   projectDetailPage: `${VERRA_REGISTRY}/app/projectDetail/VCS`, // add ID after VCS like /191
 };
+export const IS_PRODUCTION =
+  process.env.NEXT_PUBLIC_VERCEL_ENV === "production";
```
Comment (Atmosfearful) [inline lib/constants/index.ts:258]: I would prefer that `lib` stay completely agnostic about nextjs or vercel, so we will have to pass this flag down at runtime

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

**Review by ladytrekker — COMMENTED** [review]
One thing is still missing I think:
https://github.com/KlimaDAO/klimadao/blob/staging/site/next.config.js#L92-L107
=> adding Hindi to the list of production languages

Also this should have been added to this fix here?
https://github.com/KlimaDAO/klimadao/pull/728/files

File: lib/utils/getLocales/index.ts (hunk 2 above)
Comment (biwano) [reply lib/utils/getLocales/index.ts:1]: Indeed

File: lib/constants/index.ts (hunk 3 above)
Comment (biwano) [reply lib/constants/index.ts:258]: Okay

**Comment by biwano** [comment]
> One thing is still missing I think: https://github.com/KlimaDAO/klimadao/blob/staging/site/next.config.js#L92-L107 => adding Hindi to the list of production languages
> 
> I assume this should have been added to this PR here too? #728

Yes.  Introduced the async function version of the configuration module to enable dynamic imports: https://nextjs.org/docs/api-reference/next.config.js/introduction

I will make some tests on production builds this week end.

**Comment by biwano** [comment]
My production builds seemed okay.

File: lib/constants/index.ts (hunk 3 above)
Comment (ladytrekker) [inline lib/constants/index.ts:258]: I think this needs to be removed from lib still.

**Review by ladytrekker — APPROVED** [review]
Code and Preview Links look good to me.

See some minor thing below

File: lib/utils/getLocales/index.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  de: { plurals: de, time: "de-DE", label: "Deutsch" },
+  ru: { plurals: ru, time: "ru-RU", label: "Русский" },
+  "zh-CN": {
+    plurals: zh,
+    time: "zh-CN",
+    label: "中文",
+  },
+  es: { plurals: es, time: "es-ES", label: "Español" },
+  hi: { plurals: hi, time: "hi-IN", label: "हिन्दी" },
+  ko: { plurals: ko, time: "ko-KR", label: "한국어" },
```
Comment (ladytrekker) [inline lib/utils/getLocales/index.ts:28]: This is not correct IMO, korean is still only for Staging, right?
Double check with https://www.klimadao.finance/
and https://app.klimadao.finance/#/stake

**Pull request merged**